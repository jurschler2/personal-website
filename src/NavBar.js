import React from "react";
import { NavLink } from "react-router-dom";


function NavBar() {
  return (
    <nav>
      <NavLink exact to="/">JU</NavLink>
      <NavLink exact to="/about">About</NavLink>
      <NavLink exact to="/portfolio">Portfolio</NavLink>
      <NavLink exact to="/contact">Contact</NavLink>
    </nav>
  )
}

export default NavBar;