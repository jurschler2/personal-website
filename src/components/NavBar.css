/** 
 * Styling for the NavBar component:
 */

nav a {
  color: #242337;
  text-decoration: none;
  font-weight: 400;
}

nav {
  background-color: #e9e9eb;
}

nav a.active {
  color: #39384b;
  font-weight: bold;
  border-bottom: 5px;
  box-sizing: border-box;
}

nav a.active:hover {
  color: #656573 !important;
}

nav a:hover {
  text-decoration: none;
  color: #656573;
}

.personal-logo {
  font-family: Kanit;
  font-weight: 400;
  font-size: large;
  color: #242337;
}