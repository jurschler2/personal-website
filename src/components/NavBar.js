import React from "react";
import { NavLink } from "react-router-dom";

/**
 *  DESCRIPTION:
 *  PROPS: None
 *  FLOW: App => NavBar
 *  PARENT: App
 *  CHILDREN: None
 */

function NavBar() {
  return (
    <span>
      <nav>
        <NavLink exact to="/">JU</NavLink>
      </nav>
      <nav>  
        <NavLink exact to="/about">About</NavLink>
        <NavLink exact to="/portfolio">Portfolio</NavLink>
        <NavLink exact to="/contact">Contact</NavLink>
      </nav>
    </span>
  )
}

export default NavBar;