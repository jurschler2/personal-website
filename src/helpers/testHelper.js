import React from "react";
import { render } from "@testing-library/react";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";

export function renderWithRouter(children, initialPath = "") {
  const history = createMemoryHistory();
  if (initialPath) history.push(initialPath);
  return render(<Router history={history}>{children}</Router>);
}
