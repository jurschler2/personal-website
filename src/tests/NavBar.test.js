import React from "react";
import NavBar from "../components/NavBar";
import {renderWithRouter} from "../helpers/testHelper"

it("renders without crashing", () => {
  renderWithRouter(<NavBar />);
});

it("matches snapshot", () => {
  const { asFragment } = renderWithRouter(<NavBar />);
  expect(asFragment()).toMatchSnapshot();
});