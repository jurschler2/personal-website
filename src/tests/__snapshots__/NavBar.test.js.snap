// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <nav>
    <a
      aria-current="page"
      class="active"
      href="/"
    >
      JU
    </a>
    <a
      href="/about"
    >
      About
    </a>
    <a
      href="/portfolio"
    >
      Portfolio
    </a>
    <a
      href="/contact"
    >
      Contact
    </a>
  </nav>
</DocumentFragment>
`;
